File: fern/snippets/typescript/quickstart.ts
import { Composio } from "@composio/core";
import { OpenAI } from "openai";

const openai = new OpenAI();
const composio = new Composio();

const userId = "your@email.com";

const connection = await composio.toolkits.authorize(userId, "github");

console.log(
  `🔗 Please authorize access by visiting this URL:\n👉 ${connection.redirectUrl}`
);

await connection.waitForConnection();

const tools = await composio.tools.get(
  userId,
  "GITHUB_STAR_A_REPOSITORY_FOR_THE_AUTHENTICATED_USER"
);

const completion = await openai.chat.completions.create({
  model: "gpt-4o",
  messages: [
    {
      role: "user",
      content: "Can you star the composiohq/composio repository?",
    },
  ],
  tools: tools,
});

if (completion.choices[0].message.tool_calls) {
  const toolResult = await composio.provider.executeToolCall(
    userId,
    completion.choices[0].message.tool_calls[0]
  );

  console.log("✅ Tool execution result:", JSON.parse(toolResult));
}

File: fern/snippets/python/quickstart.py
import os
from composio_core import Composio

userId = "your@email.com"

composio = Composio(api_key=os.environ.get("COMPOSIO_API_KEY"))

print("🚀 Starting GitHub authentication for user:", userId)

connection = composio.toolkits.authorize(userId, "github")

print("🔗 Please authorize access by visiting this URL:")
print("👉", connection.redirect_url)

print("⏳ Waiting for you to complete the authorization in your browser...")
connection.wait_for_connection()
print("✅ Authorization complete! 🎉")

print("🌟 Starring the composiohq/composio repository on GitHub...")

res = composio.tools.execute(
    "GITHUB_STAR_A_REPOSITORY_FOR_THE_AUTHENTICATED_USER",
    {"userId": userId, "arguments": {"owner": "composiohq", "repo": "composio"}},
)

print("⭐️ Repository starred! Here is the response:")
print(res)
